src/scripts/modules/media/body/body.coffee

```diff
@@ -152,6 +152,31 @@ define (require) ->
             # NOTE: `.toggleClass()` explicitly requires a `false` (not falsy) 2nd argument
             $el.toggleClass('ui-has-child-title', $title.length > 0)
 
+          # Create subtitles in each chapter with corresponding ids as their href
+          total_subtitle = $temp.find("[data-depth='1']")
+          total_id = total_subtitle.children('[data-type="title"]')
+          id_length = total_id.length
+          i = 0
+          slash_index = window.location.href.lastIndexOf("/")+1
+          real_href = window.location.pathname
+          if real_href.indexOf("#")!= -1
+            hash_index = real_href.indexOf("#")-1
+            real_href = real_href.substring(0,hash_index)
+          location = $('.table-of-contents [href$="'+real_href+'"]').children(".title")
+          location2 = $('.table-of-contents [href$="'+real_href+'"]').children(".chapter-number")
+          # Keep adding subtitles until all subtitles are added
+          if location.children("li").length < id_length
+            while i < id_length
```

Comment:
Hi, Phil I think this will actually cause a problem because if I click a chapter several times, it will create many repetitive lists.